### PR TITLE
fix: bail validation and refactor create button labels

### PR DESCRIPTION
**Pull request by dni**
- button label is now way easier to work with
- validation only happens if there is a amount set at all

### CODE REVIEW

File: src/components/CreateButton.tsx [hunk 1]
```diff
@@ -34,42 +45,108 @@ export const CreateButton = () => {
...
+                        params: {
+                            amount: formatAmount(
+                                BigNumber(lessThanMin ? minimum() : maximum()),
+                                denomination(),
+                            ),
+                            denomination: denomination(),
+                        },
+                    });
+                    return;
+                } else {
```
Comment (michael1011) [inline src/components/CreateButton.tsx:114]: No need for else if you always return in the previous `if`s

File: src/components/CreateButton.tsx [hunk 2]
```diff
@@ -209,7 +286,7 @@ export const CreateButton = () => {
             id="create-swap-button"
             data-testid="create-swap-button"
             class={buttonClass()}
-            disabled={buttonDisable() || !online()}
+            disabled={buttonDisable()}
```
Comment (michael1011) [inline src/components/CreateButton.tsx:289]: image
Invoices of the wrong network have a yellow button that can't be clicked. How about setting the class of the button when the disabled status is changed?

**Review by michael1011 — CHANGES_REQUESTED** [review]
There are no tests

**Comment by dni** [comment]
closes #443

**Comment by dni** [comment]
closes #353

File: tests/components/CreateButton.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        signals.setAmountValid(true);
+        signals.setAddressValid(true);
+        signals.setReverse(true);
+        const btn = (await screen.findByText(
+            i18n.en.create_swap,
+        )) as HTMLButtonElement;
+        expect(btn).not.toBeUndefined();
+        expect(btn.disabled).toBeFalsy();
+        signals.setAddressValid(false);
+        expect(btn.disabled).toBeTruthy();
```
Comment (michael1011) [inline tests/components/CreateButton.spec.tsx:94]: Let's also assert the text changed to address invalid

**Review by michael1011 — APPROVED** [review]

**Review by kilrau — CHANGES_REQUESTED** [review]
https://github.com/BoltzExchange/boltz-web-app/issues/443 seems to be fixed, couldn't get web app to resolve my lightning address with invalid/empty amounts

https://github.com/BoltzExchange/boltz-web-app/issues/353 for empty amounts the "Please provide a valid LN invoice, LNAddress or LNURL" should be shown too if I enter sth invalid: 
image
Same for Normal Swaps with empty amounts, it should show "Invalid BTC address"
image

**Review by kilrau — CHANGES_REQUESTED** [review]
Let's fix error prios:
- let's stick with "Minimum amount is" as default state when loading the page, just as current master. But NEVER circle invoice/address field red if it's empty. Only when it's an invalid lnurl/lnaddress format. Grey button with error message is clear enough. Yes WebLN can be used to paste an invoice with amount and users will click the button if it's there even though the message says enter an amount. Accepted edge case trade-off.
image
- amount error supersedes all other errors, wrong ln address/lnurl show only if valid amount is set, test: amount error stays when I enter wrong lnurl/address
- invoice errors like 0 amount can show as-is as they are popups
- re-run amount validation when I manipulate amounts that passed validation before
- re-run lnurl/lnaddress/invoice/address validationwhen I manipulate an input that passed validation before

**Review by kilrau — CHANGES_REQUESTED** [review]
One minor things left:

- [ ] you are still showing red if the address input field is empty. Please show grey in case it's empty. Shouldn't have an error state if I'm just done punching in amounts
image

**Review by kilrau — APPROVED** [review]
LGTM

**Review by michael1011 — APPROVED** [review]
